packages/discovery-react-components/src/components/SearchResults/SearchResults.tsx

```diff
@@ -77,6 +77,10 @@ export interface SearchResultsProps {
    * callback function from the component for sending document
    */
   onSelectResult?: (document: { document: DiscoveryV2.QueryResult }) => void | undefined;
+  /**
+   * Used for tracking analytics
+   */
+  trackEventSpelling?: (payload: { eventName: any; eventProps: { 'custom.term': string } }) => void;
```

Comment:
since this is a complex compound component, i could see this being called `onSpellingClick?: (e: SyntheticEvent<EventTarget>)`